.github/workflows/scripts/shared/verify_changes_update_version.sh

```diff
@@ -1,27 +1,48 @@
 # Verifies changes to directory have an updated version
 
 if [ -z "$1" ] || [ -z "$2" ]; then
-  echo "Usage: sh $0 [potential_git_tag] <path_to_directory>"
+  echo "Usage: sh $0 <potential_git_tag> <path_to_directory> [extra_regex_excludes]"
   exit 1
 fi
 
-set -eux
+set -eu
 
 POTENTIAL_GIT_TAG=$1
 DIRECTORY=$2
+ADDITIONAL_DIFF_IGNORE=${3:-}
+
+GREP_FILTER_STDERR_OUTPUT="/tmp/diff_grep_filter_stderr"
 
 main() {
   # github actions job does not fetch other git objects by default
   git fetch origin $GITHUB_BASE_REF
   git fetch --tags
 
-  changes_in_pr=$(git diff origin/$GITHUB_BASE_REF -- $DIRECTORY)
-  if [ -n "$changes_in_pr" ]; then
-    echo "Found changes between current branch and $GITHUB_BASE_REF"
+  filtered_changes=$(git diff --name-only origin/$GITHUB_BASE_REF -- $DIRECTORY | grep -E -v \
+                                                                            -e 'test/'  \
+                                                                            -e 'requirements-dev\.txt$' \
+                                                                            -e '\.md$' \
+                                                                            $ADDITIONAL_DIFF_IGNORE \
+                                                                            2> $GREP_FILTER_STDERR_OUTPUT \
+                                                                            || true)
```

Comment:
📓 Should we have a small comment saying why we have || true? Since its needed incase no changes are found.

Reply:
It isn't small :smile: , but I added an explanatory comment.